internal/env/parser.go

```
@@ -112,6 +115,19 @@ func parseServerConfig(cfg *server.Config, altPrefix string, opts ...env.Options
 		if (cfg.Port == "" || cfg.Port == copyConfig.Port) && altCfg.Port != "" {
 			cfg.Port = altCfg.Port
 		}
+		if cfg.TracingURL == "" && altCfg.TracingURL != "" {
+			cfg.TracingURL = altCfg.TracingURL
+		}
+		if cfg.InstanceID == "" && altCfg.InstanceID != "" {
+			cfg.InstanceID = altCfg.InstanceID
+		}
+		if cfg.InstanceID == "" && altCfg.InstanceID == "" {
+			instanceID, err := uuid.New().ID()
+			if err != nil {
+				log.Fatalf("InstanceID not generated: %s", err)
```

Return an error, rather than adding a logger here.